miden-lib/src/transaction/inputs.rs

```diff
@@ -25,7 +25,11 @@ impl ToTransactionKernelInputs for PreparedTransaction {
         let account = self.account();
         let stack_inputs = TransactionKernel::build_input_stack(
             account.id(),
-            if account.is_new() { None } else { Some(account.hash()) },
+            if account.is_new() {
+                Digest::default()
+            } else {
+                account.hash()
+            },
```

Comment:
We repeat this code snippet several times. I wonder if it should go in a small method implemented on the `Account` struct. Maybe something like `proof_input_hash()` or `proof_init_hash()`? This would also be a good place to document it.

Reply:
I agree with the DRY idea, but this seems very specific to the problem we are facing now no?

Would we use this method in other places?

Reply:
Not sure if we'll use it in other places (or at least I can't think of any now). But we do use it in a couple of places now, and my concern is that if something changes in the future, it would be easy to miss one of these. Having a separate method prevents this.

Reply:
```rust
    /// Returns hash of account used as input for proofs
    pub fn proof_input_hash(&self) -> Digest {
        if self.is_new() {
            Digest::default()
        } else {
            self.hash()
        }
    }
```

Reply:
Could you explain a bit more in the docs why we do it this way?

Reply:
```rust
    /// Returns hash of account used as initial input for proofs.
    ///
    /// We handle old and new accounts differently when generating proofs.
    /// For new accounts we do not need to pass the initial account hash,
    /// hence using `Digest::default()`
    pub fn proof_init_hash(&self) -> Digest {
        if self.is_new() {
            Digest::default()
        } else {
            self.hash()
        }
    }
 ```

This is how I have mostly understood it from your comment here: https://github.com/0xPolygonMiden/miden-base/issues/454#issuecomment-1934610143

Reply:
Yep, that look right. I would maybe change the wording a bit - e.g., to something like this:
```Rust
/// Returns hash of this account as used for the initial account state hash in transaction
/// proofs.
///
/// For existing accounts, this is exactly the same as [Account::hash()], however, for new
/// accounts this value is set to [ZERO; 4]. This is because when a transaction is executed
/// agains a new account, public input for the initial account state is set to [ZERO; 4] to
/// distinguish new accounts from existing accounts. The actual hash of the initial account
/// state (and the initial state itself), are provided to the VM via the advice provider.
```